PayPalWebPayments/src/main/java/com/paypal/android/paypalwebpayments/PayPalWebCheckoutClient.kt

```diff
@@ -78,69 +56,74 @@ class PayPalWebCheckoutClient internal constructor(
-            try {
-                val browserSwitchOptions = browserSwitchHelper.configurePayPalBrowserSwitchOptions(
-                    request.orderId,
-                    coreConfig,
-                    request.fundingSource
-                )
-                browserSwitchClient.start(activity, browserSwitchOptions)
-            } catch (e: PayPalSDKError) {
-                deliverFailure(APIClientError.clientIDNotFoundError(e.code, e.correlationId))
-            }
+        analyticsService.sendAnalyticsEvent("paypal-web-payments:started", request.orderId)
+        payPalWebLauncher.launchPayPalWebCheckout(activity, request)?.let { launchError ->
+            notifyWebCheckoutFailure(launchError)
         }
     }
 
-    internal fun handleBrowserSwitchResult() {
-        browserSwitchResult = browserSwitchClient.deliverResult(activity)
-        listener?.also {
-            browserSwitchResult?.also { result ->
-                when (result.status) {
-                    BrowserSwitchStatus.SUCCESS -> deliverSuccess()
-                    BrowserSwitchStatus.CANCELED -> deliverCancellation()
-                }
-            }
+    /**
+     * Vault PayPal as a payment method. Result will be delivered to your [PayPalWebVaultListener].
+     *
+     * @param request [PayPalWebVaultRequest] for vaulting PayPal as a payment method
+     */
+    fun vault(request: PayPalWebVaultRequest) {
+        payPalWebLauncher.launchPayPalWebVault(activity, request)?.let { launchError ->
+            notifyVaultFailure(launchError)
         }
     }
 
-    private fun deliverSuccess() {
-        if (browserSwitchResult?.deepLinkUrl != null && browserSwitchResult?.requestMetadata != null) {
-            val webResult = PayPalDeepLinkUrlResult(
-                browserSwitchResult?.deepLinkUrl!!,
-                browserSwitchResult?.requestMetadata!!
-            )
-            if (!webResult.orderId.isNullOrBlank() && !webResult.payerId.isNullOrBlank()) {
-                deliverSuccess(
-                    PayPalWebCheckoutResult(
-                        webResult.orderId,
-                        webResult.payerId
-                    )
-                )
-            } else {
-                deliverFailure(PayPalWebCheckoutError.malformedResultError)
+    internal fun handleBrowserSwitchResult() {
+        payPalWebLauncher.deliverBrowserSwitchResult(activity)?.let { status ->
+            when (status) {
+                is PayPalWebStatus.CheckoutSuccess -> notifyWebCheckoutSuccess(status.result)
+                is PayPalWebStatus.CheckoutError -> notifyWebCheckoutFailure(status.error)
+                is PayPalWebStatus.CheckoutCanceled -> notifyWebCheckoutCancelation(status.orderId)
+                is PayPalWebStatus.VaultSuccess -> notifyVaultSuccess(status.result)
+                is PayPalWebStatus.VaultError -> notifyVaultFailure(status.error)
+                PayPalWebStatus.VaultCanceled -> notifyVaultCancelation()
```

Comment:
```suggestion
               is PayPalWebStatus.VaultCanceled -> notifyVaultCancelation()
```
surprised this compiled - is the "is" not required? 🤔

Reply:
Yeah in this case `PayPalWebStatus.VaultCanceled` is an `object` type so it's equivalent to `status === PayPalWebStatus.VaultCanceled` in the `when` statement.